fdbserver/CommitProxyServer.actor.cpp

```diff
@@ -1584,6 +1587,27 @@ void determineCommittedTransactions(CommitBatchContext* self) {
 	}
 }
 
+// Return true if a single-key mutation is associated with a valid tenant id or a system key
+bool validTenantAccess(MutationRef m) {
+	if (isSystemKey(m.param1))
+		return true;
+
+	if (isSingleKeyMutation((MutationRef::Type)m.type)) {
+		auto tenantId = TenantAPI::extractTenantIdFromMutation(m);
+		// throw exception for invalid raw access
+		if (tenantId == TenantInfo::INVALID_TENANT) {
+			return false;
+		}
+	} else {
+		// For clear range, we allow raw access
+		ASSERT_EQ(m.type, MutationRef::Type::ClearRange);
+		auto beginTenantId = TenantAPI::extractTenantIdFromKeyRef(m.param1);
+		auto endTenantId = TenantAPI::extractTenantIdFromKeyRef(m.param2);
+		CODE_PROBE(beginTenantId != endTenantId, "Clear Range raw access or cross multiple tenants");
+	}
+	return true;
+}
+
```

Comment:
This function now exists at line 1076, and since it doesn't look like you are using it for locking I think we can just delete it.

```suggestion
```